### PR TITLE
add interface for inject

### PR DESCRIPTION
**Pull request by songyang-lotusFlare**

### CODE REVIEW

File: app/src/main/java/ph/com/globe/gomo/ui/login/LoginFragment.kt [hunk 1]
```diff
@@ -25,7 +25,7 @@ class LoginFragment : BaseFragment() {
     private lateinit var binding: FragmentLoginBinding
 
     @Inject
-    lateinit var profileModuleManager: ProfileModuleManager
+    lateinit var profileModuleManager: IProfileDomainManager
```
Comment (songyang-lotusFlare) [inline app/src/main/java/ph/com/globe/gomo/ui/login/LoginFragment.kt]: need to move to `ViewModel`

File: domain/src/main/java/ph/com/globe/domain/profile/IProfileDomainManager.kt [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+
+import kotlinx.coroutines.flow.Flow
+import ph.com.globe.common.LfResult
+import ph.com.globe.common.errors.profile.GetGomoUserError
+import ph.com.globe.model.profile.domin_models.GomoUserModel
+
+interface IProfileDomainManager {
+
+    suspend fun getGomoUserCase(): Flow<LfResult<GomoUserModel?, GetGomoUserError>>
+}
```
Comment (songyang-lotusFlare) [inline domain/src/main/java/ph/com/globe/domain/profile/IProfileDomainManager.kt]: add an new line

File: domain/src/main/java/ph/com/globe/domain/profile/di/ProfileDomainModule.kt [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+import ph.com.globe.domain.profile.IProfileDomainManager
+import ph.com.globe.domain.profile.ProfileDomainManager
+
+@Module
+@InstallIn(SingletonComponent::class)
+internal interface ProfileDomainModule {
+
+    @Binds
+    fun providerProfileDomainManager(profileDomainManager: ProfileDomainManager): IProfileDomainManager
+}
```
Comment (songyang-lotusFlare) [inline domain/src/main/java/ph/com/globe/domain/profile/di/ProfileDomainModule.kt]: new line

File: domain/src/main/java/ph/com/globe/domain/profile/IProfileDomainManager.kt [hunk 4]
```diff
@@ -0,0 +1,11 @@
+package ph.com.globe.domain.profile
+
+import kotlinx.coroutines.flow.Flow
+import ph.com.globe.common.LfResult
+import ph.com.globe.common.errors.profile.GetGomoUserError
+import ph.com.globe.model.profile.domin_models.GomoUserModel
+
+interface IProfileDomainManager {
+
```
Comment (Lotzzz) [inline domain/src/main/java/ph/com/globe/domain/profile/IProfileDomainManager.kt]: remove the empty line

File: domain/src/main/java/ph/com/globe/domain/profile/ProfileDomainManager.kt [hunk 5]
```diff
@@ -1,17 +1,18 @@
...
 import ph.com.globe.common.errors.profile.GetGomoUserError
 import javax.inject.Inject
 
 //Finally injected into the app layer class
-class ProfileModuleManager @Inject constructor(
+internal class ProfileDomainManager @Inject constructor(
     //useCase ...
     private val gomoUserUseCase: GomoUserUseCase
-) {
+): IProfileDomainManager {
```
Comment (Lotzzz) [inline domain/src/main/java/ph/com/globe/domain/profile/ProfileDomainManager.kt:7]: remove the empty line

File: domain/src/main/java/ph/com/globe/domain/profile/di/ProfileDomainModule.kt [hunk 6]
```diff
@@ -0,0 +1,16 @@
...
+import dagger.Module
+import dagger.hilt.InstallIn
+import dagger.hilt.components.SingletonComponent
+import ph.com.globe.domain.profile.IProfileDomainManager
+import ph.com.globe.domain.profile.ProfileDomainManager
+
+@Module
+@InstallIn(SingletonComponent::class)
+internal interface ProfileDomainModule {
+
```
Comment (Lotzzz) [inline domain/src/main/java/ph/com/globe/domain/profile/di/ProfileDomainModule.kt]: remove the empty line

File: domain/src/main/java/ph/com/globe/domain/profile/usecase/GomoUserUseCase.kt [hunk 7]
```diff
@@ -15,7 +15,7 @@ import ph.com.globe.data.db.IRepositoryManager
 import javax.inject.Inject
 
 
-class GomoUserUseCase @Inject constructor(repoManager: IRepositoryManager) {
+internal class GomoUserUseCase @Inject constructor(repoManager: IRepositoryManager) {
```
Comment (Lotzzz) [inline domain/src/main/java/ph/com/globe/domain/profile/usecase/GomoUserUseCase.kt]: remove the empty line

File: common/src/main/java/ph/com/globe/common/errors/GeneralError.kt [hunk 8]
```diff
@@ -1,11 +1,7 @@
...
 
 sealed class GeneralError {
-
     object NotLoggedIn: GeneralError()
-
     object General: GeneralError()
-
     data class Other(val error: NetworkError): GeneralError()
-
 }
```
Comment (Lotzzz) [inline common/src/main/java/ph/com/globe/common/errors/GeneralError.kt]: Please add a new line

File: data/src/main/AndroidManifest.xml [hunk 9]
```diff
@@ -1,5 +1,6 @@
 <?xml version="1.0" encoding="utf-8"?>
 <manifest xmlns:android="http://schemas.android.com/apk/res/android"
     package="ph.com.globe.data">
-
+    <uses-permission android:name="android.permission.ACCESS_NETWORK_STATE" />
+    <uses-permission android:name="android.permission.ACCESS_WIFI_STATE" />
 </manifest>
```
Comment (Lotzzz) [inline data/src/main/AndroidManifest.xml]: Please add a new line

**Review by Lotzzz — APPROVED** [review]

**Pull request merged**